Tic_tac_toe/js/game.js

```diff
@@ -0,0 +1,162 @@
+import { gameOptions } from "./constants.js";
+
+export class Game {
+  players = [];
+
+  constructor() {
+    this.drawEl = document.querySelector(".game__info-draw-value");
+
+    this.fieldEl = document.querySelector(".field");
+    this.addCells();
+
+    this.moveEl = document.querySelector(".game__move-value");
+  }
+
+  play() {
+    this.setDefault();
+
+    this.fieldEl.addEventListener("click", (evt) => {
+      if (evt.target.dataset.value) {
+        return;
+      }
+
+      if (!evt.target.classList.contains("field__cell")) {
+        return;
+      }
+
+      evt.target.dataset.value = this.activePlayerInd === 0 ? "cross" : "nule";
+
+      this.updateVirtualField(evt.target);
+
+      //Кто-то выиграл
+      if (this.checkWinner(this.virtualField)) {
+        this.activePlayer.win();
+        this.clearField();
+      }
+
+      //Ничья
+      if (this.checkDraw()) {
+        this.draw++;
+        this.updateDraws();
+        this.clearField();
+      }
+
+      this.changeActivity();
+      this.moveEl.textContent = this.activePlayer.getName();
+    });
+  }
+
+  repeat() {
+    this.setDefault();
+  }
+
+  setDefault() {
+    this.draw = 0;
+    this.updateDraws();
+
+    this.clearField();
+
+    this.activePlayerInd = 0;
+    this.activePlayer = this.players[0];
+
+    this.updateMove();
+
+    this.players.forEach((player) => player.setDefault());
+  }
+
+  clearField() {
+    Array.from(this.fieldEl.children).forEach((cell) => {
+      cell.dataset.value = "";
+    });
+
+    this.fillVirtualField();
+  }
+
+  checkWinner(arr) {
+    function Eq(a, b, c) {
+      return a !== 0 && a == b && a == c;
+    }
+    for (let i = 0; i < 3; i++) {
```

Comment:
will this check work, if user will change `gameOptions.SIZE` to 4 or 5?

Reply:
No, I need to write new algoritm to find winners, if size will be different of 3